sysapi/sysapi/Tss2_Sys_CreatePrimary.c

```diff
@@ -30,117 +30,152 @@
+    if (rval)
+        return rval;
+
+    rval = TPM2B_DIGEST_Unmarshal(SYS_CONTEXT->tpmOutBuffPtr,
+                                  SYS_CONTEXT->maxResponseSize,
+                                  &SYS_CONTEXT->nextData, creationHash);
+    if (rval)
+        return rval;
+
+    rval = TPMT_TK_CREATION_Unmarshal(SYS_CONTEXT->tpmOutBuffPtr,
+                                      SYS_CONTEXT->maxResponseSize,
+                                      &SYS_CONTEXT->nextData, creationTicket);
+    if (rval)
+        return rval;
+
+    rval = TPM2B_NAME_Unmarshal(SYS_CONTEXT->tpmOutBuffPtr,
+                                SYS_CONTEXT->maxResponseSize,
+                                &SYS_CONTEXT->nextData, name);
+    return rval;
 }
 
 TPM_RC Tss2_Sys_CreatePrimary(
     TSS2_SYS_CONTEXT *sysContext,
-    TPMI_RH_HIERARCHY	primaryHandle,
+    TPMI_RH_HIERARCHY primaryHandle,
     TSS2_SYS_CMD_AUTHS const *cmdAuthsArray,
-    TPM2B_SENSITIVE_CREATE	*inSensitive,
-    TPM2B_PUBLIC	*inPublic,
-    TPM2B_DATA	*outsideInfo,
-    TPML_PCR_SELECTION	*creationPCR,
-    TPM_HANDLE	*objectHandle,
-    TPM2B_PUBLIC	*outPublic,
-    TPM2B_CREATION_DATA	*creationData,
-    TPM2B_DIGEST	*creationHash,
-    TPMT_TK_CREATION	*creationTicket,
-    TPM2B_NAME	*name,
-    TSS2_SYS_RSP_AUTHS *rspAuthsArray
-    )
+    TPM2B_SENSITIVE_CREATE *inSensitive,
+    TPM2B_PUBLIC *inPublic,
+    TPM2B_DATA *outsideInfo,
+    TPML_PCR_SELECTION *creationPCR,
+    TPM_HANDLE *objectHandle,
+    TPM2B_PUBLIC *outPublic,
+    TPM2B_CREATION_DATA *creationData,
+    TPM2B_DIGEST *creationHash,
+    TPMT_TK_CREATION *creationTicket,
+    TPM2B_NAME *name,
+    TSS2_SYS_RSP_AUTHS *rspAuthsArray)
 {
-    TSS2_RC     rval = TPM_RC_SUCCESS;
+    TSS2_RC rval;
 
-    if( creationPCR == NULL  )
-	{
-		return TSS2_SYS_RC_BAD_REFERENCE;
-	}
+    if (!sysContext || !creationPCR)
+        return TSS2_SYS_RC_BAD_REFERENCE;
 
-    rval = Tss2_Sys_CreatePrimary_Prepare( sysContext, primaryHandle, inSensitive, inPublic, outsideInfo, creationPCR );
+    rval = Tss2_Sys_CreatePrimary_Prepare(sysContext, primaryHandle, inSensitive,
+                                          inPublic, outsideInfo, creationPCR);
+    if (rval)
+        return rval;
 
-    if( rval == TSS2_RC_SUCCESS )
-    {
-        rval = CommonOneCall( sysContext, cmdAuthsArray, rspAuthsArray );
+    rval = CommonOneCall(sysContext, cmdAuthsArray, rspAuthsArray);
 
-        if( rval == TSS2_RC_SUCCESS )
-        {
-            rval = Tss2_Sys_CreatePrimary_Complete( sysContext, objectHandle, outPublic, creationData, creationHash, creationTicket, name );
-        }
-    }
+    if (rval)
+        return rval;
 
+    rval = Tss2_Sys_CreatePrimary_Complete(sysContext, objectHandle, outPublic,
+                                           creationData, creationHash,
+                                           creationTicket, name);
     return rval;
-}
 
+}
```

Comment:
Whitespace between return and the closing parenthesis departs from the norm.

Reply:
Do you mean blank line?

Reply:
yes